keps/487-kubectl-plugin-listing/README.md

```diff
@@ -0,0 +1,316 @@
+        Security Context:
+        Termination Grace Period Seconds:  30
+  Priority:                                0
+  Priority Class Source:                   
+  Queue Name:                              lq-team-a
+
+```
+
+In order to make kueue CLI simple, we are going to have only one abstraction for jobs specific information — workload. Currently the `describe` command will mostly duplicate the `kubectl describe` command with few modifications:
+`--status` option will display Workload.Status field and state of Workload.
+There is no `State` field in Workload, but it could be identified from other fields. The `Workload.Status` has an `Admitted` field, in case of pending workload it is empty, so from the user point of view it's not clear what the state of the Workload is. I suggest adding the state information that displays `Admitted` if the `Status.Admitted` is not empty and `Pending` if the `Status.Admitted` is empty.
+The `--spec` option will display `Workload.Spec` information.
+
+For ClusterQueue the Status field might look confusing. For example for misconfigured ClusterQueur `Status.Conditions[-1].Type` have value `Active` and `Status.Conditions[-1].Status` have value `False`. I suggest including a state field that will display `Error` if `Status.Conditions.Status` is False and `Active` if `Status.Conditions.Status` is `True` and `Status.Conditions.Type` is `Active`. Later, when we'll support other types, the logic will probably stay the same, but it should be double checked.
+
+Since the logic for the `State` field is complex, we need to make sure it's well tested and be able to identify when the logic should be updated. The proposal is to have a dedicated test for `State` next to the API definition.
+
+### List
+
+```bash
+$kubectl kueue worlkoads list --clusterqueue=cq
+| Namespace | Name     |  Queue  | ClusterQueue | Pods |   State   | Age
+|-----------|----------|---------|--------------|------|-----------|-----
+| default   | wl-1     | queue-1 |   cq         |   2  | Admitted  | 2h
+| default   | wl-2     | queue-2 |   cq         |   5  | Pending   | 2h
+
+
+$kubectl kueue clusterqueue list
+| Name             | Cohort   |  State  | PENDING WORKLOADS
+|------------------|----------|---------|-------------------
+| cluster-queue-1  | cohort   | Active  | 2  
+| cluster-queue-2  |          | Error   | 0    
+
+```
+
+The list command will use kubectl get command with CustumnColumntPrinter (to include ClusterQueue column) and with attached column `State`, that was described earlier. For Workloads we'll use sorting for pending workloads that are on the same ClusterQueue based on their position on the queue. Since ClusterQueue is not namespaced object, I suggest using all namespaces by default.
+The command will support --state, --cohort, --clusterqueue, --namespace flags when applicable.
+
+### Stop
+
+```bash
+$kubectl kueue worlkoad stop workload-name --clusterqueue cluster-queue-name
```

Comment:
Stop is ambiguous.  Does it kill the job with the failed status, or just removes it from the queue for immediate readmission? How to "unstop" a workload?

Reply:
I see, removed it since it's out of scope for this kep